### ISSUE TITLE
[HOLD for payment 2023-09-21] [$500]  [Distance] - There is NO animation of the transition when I have entered start and finish points

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#26723]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Issue found when executing PR** https://github.com/Expensify/App/pull/26610
## Action Performed:
1. Open app on small screen (height = +- 400px )
2. Go to FAB -> Request money -> Distance
3. Enter start and finish points
4. Click "Add stope"

## Expected Result:
Should be an animation of the transition of the list elements to the end

## Actual Result:
There is NO animation of the transition of the list elements to the end

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** 1.3.63-0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/6ec3a461-6ef3-440e-83e6-ff01d71b19e8

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0116cd2b0228d37197</li>
        <li>Upwork Job ID: 1699803735595442176</li>
        <li>Last Price Increase: 2023-09-07</li>
<li>Automatic offers: </li>
<ul>
<li>jjcoffee | Reviewer | 26572301</li>
<li>bernhardoj | Contributor | 26572302</li>
</ul></li>
    </ul>
</details>

**Comment by jscardona12** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Distance - There is NO animation of the transition when I have entered start and finish points.
### What is the root cause of that problem?
The line https://github.com/Expensify/App/blob/8150ce011bcc4c685cb08627681a6fb1fe4276c7/src/components/DistanceRequest.js#L174 scrollToEnd does not work. 
### What changes do you think we should make in order to solve the problem?
We should modify https://github.com/Expensify/App/blob/8150ce011bcc4c685cb08627681a6fb1fe4276c7/src/components/DistanceRequest.js#L174 and set it as `scrollViewRef.current.scrollTo({y:height, animated: true});` 
### What alternative solutions did you explore? (Optional)

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
There is NO animation of the transition when I have entered start and finish points

### What is the root cause of that problem?
https://github.com/Expensify/App/blob/6b1a6679042c4fa9bb628360e028f93a5f27b2f2/src/components/DistanceRequest.js#L173-L175

scrollToEnd function is not called because numberOfWaypoints = numberOfPreviousWaypoints

We are using usePrevious hook to save the previous waypoint. When this component is re-render 2 times, for example, waypoints are 2. The first re-rendering Waypoints is updated to 3 and previousWaypoints is updated to 2, the second re-rendering Waypoints is not updated but the previousWaypoints still updated to 3 (the value of Waypoints in the first time)

### What changes do you think we should make in order to solve the problem?
https://github.com/Expensify/App/blob/6b1a6679042c4fa9bb628360e028f93a5f27b2f2/src/components/DistanceRequest.js#L173-L175

In here, we can remove the condition numberOfWaypoints > numberOfPreviousWaypoints
It still works well

https://github.com/Expensify/App/assets/141406735/7d327203-ef68-47d3-ad65-8a0759406cb6



### What alternative solutions did you explore? (Optional)
We only update previousWaypoints if waypoints changed

**Comment by puneetlath** [comment]
@jjcoffee thoughts on the proposals so far?

**Comment by jjcoffee** [comment]
@jscardona12's [proposal](https://github.com/Expensify/App/issues/26723#issuecomment-1705716631) lacks enough detail to fully review - there's no "why" in either the RCA or the proposed solution. Maybe the solution works (as the video demonstrates), but we'd want to be sure we're actually fixing a problem, not just making something work without understanding what we're doing.

@DylanDylann's [proposal](https://github.com/Expensify/App/issues/26723#issuecomment-1706191212) has the correct RCA (`numberOfPreviousWaypoints` isn't always what we'd expect), but misses a little bit on what actually causes the re-render. The solution seems decent, but has the downside that we'll scroll to the bottom on any `onContentSizeChange`, e.g. if you remove a waypoint you'll also be scrolled to the bottom which isn't necessarily expected behaviour (and probably why the `numberOfWaypoints` condition was added).

@bernhardoj's [proposal](https://github.com/Expensify/App/issues/26723#issuecomment-1707955572) - love the level of detail and clarity here! The RCA is correct too, but also explains _why_ the behaviour happens only once you add the start and finish points. I agree with him that we always want to scroll to the bottom when a new waypoint is added, regardless of `ScrollView` content height (so we can just ignore it). I think I prefer the alternative solution as the delay when waiting for the `Onyx` promise to resolve feels a bit too long and is unnecessary. The alternative solution also feels cleaner than slapping a scroll right onto the `addStop` call.

To summarise, happy to go with @bernhardoj's alternative [solution](https://github.com/Expensify/App/issues/26723#issuecomment-1707955572)!

:ribbon::eyes::ribbon: C+ reviewed

**Comment by arosiclair** [comment]
@bernhardoj's alternate solution LGTM too 👍

**Comment by jjcoffee** [comment]
- [x] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/26610
- [x] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/26610#issuecomment-1723339699
- [x] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/A - just needed more thorough testing**
 - [x] Determine if we should create a regression test for this bug. **No - not impactful enough, more of a "nice to have"**
- [x] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again. **N/A**

**Comment by puneetlath** [comment]
All paid. Thanks everyone!
